frontend/src/app/components/incoming-transactions-graph/incoming-transactions-graph.component.ts

```diff
@@ -72,7 +75,96 @@ export class IncomingTransactionsGraphComponent implements OnInit, OnChanges, On
     this.isLoading = false;
   }
 
+  /// calculate the moving average of this.data.series[0] based on MAWindowSize
+  calculateMA(): number[][] {
+    let ma: number[][] = [];
+    let sum = 0;
+    let i = 0;
+    let len = this.data.series[0].length;
+    let counter = 0;
+
+    ma = this.data.series[0].map((item: number[]) => {
+      return [item[0], 0];
+    });
+
+    //calculate the moving average
+    for (i = 0; i < len; i++) {
+      sum += this.data.series[0][i][1];
+      if (i >= this.MAWindowSize) {
+        sum -= this.data.series[0][i - this.MAWindowSize][1];
+        ma[i][1] = sum / this.MAWindowSize;
+      } else {
+        ma[i][1] = sum / (i + 1);
+      }
+      ma[i][0] = this.data.series[0][i][0];
```

Comment:
this section might be easier to read if we assign `this.data.series[0]` to a convenience variable at the top of the function (or even pass it in as an argument like `calculateMA(series: number[][])`)

Reply:
Love the comments, thanks mononaut! I agree with the feedback, I'll make the changes and update! 

I was even thinking of adding a window selection option for the user so they could choose how much of a smooth MA they would like to see, but still have some presets for the different time frames like you suggest. 

The center moving average definitely makes more sense for visualization (trailing moving average is more for forecasting). I'll make that change too!